Public/Src/Tools/SymbolDaemon/SymbolDaemon.cs

```diff
@@ -0,0 +1,424 @@
+
+        internal static readonly StrOption SymbolRequestNameOption = RegisterSymbolConfigOption(new StrOption("name")
+        {
+            ShortName = "n",
+            HelpText = "Request name",
+            IsRequired = true,
+        });
+
+        internal static readonly UriOption ServiceEndpoint = RegisterSymbolConfigOption(new UriOption("service")
+        {
+            ShortName = "s",
+            HelpText = "Symbol service endpoint URI",
+            IsRequired = true,
+        });
+
+        internal static readonly IntOption RetentionDays = RegisterSymbolConfigOption(new IntOption("retentionDays")
+        {
+            ShortName = "rt",
+            HelpText = "Symbol retention time in days",
+            IsRequired = false,
+            DefaultValue = (int)SymbolConfig.DefaultRetention.TotalDays,
+        });
+
+        internal static readonly IntOption HttpSendTimeoutMillis = RegisterSymbolConfigOption(new IntOption("httpSendTimeoutMillis")
+        {
+            HelpText = "Timeout for http requests",
+            IsRequired = false,
+            DefaultValue = (int)SymbolConfig.DefaultHttpSendTimeout.TotalMilliseconds,
+        });
+
+        internal static readonly BoolOption EnableTelemetry = RegisterSymbolConfigOption(new BoolOption("enableTelemetry")
+        {
+            ShortName = "t",
+            HelpText = "Verbose logging",
+            IsRequired = false,
+            DefaultValue = SymbolConfig.DefaultEnableTelemetry,
+        });
+
+        internal static SymbolConfig CreateSymbolConfig(ConfiguredCommand conf)
+        {
+            return new SymbolConfig(
+                requestName: conf.Get(SymbolRequestNameOption),
+                serviceEndpoint: conf.Get(ServiceEndpoint),
+                retention: TimeSpan.FromDays(conf.Get(RetentionDays)),
+                httpSendTimeout: TimeSpan.FromMilliseconds(conf.Get(HttpSendTimeoutMillis)),
+                verbose: conf.Get(Verbose),
+                enableTelemetry: conf.Get(EnableTelemetry),
+                logDir: conf.Get(LogDir));
+        }
+
+        private static Client CreateClient(string serverMoniker, IClientConfig config)
+        {
+            return serverMoniker != null
+                ? Client.Create(serverMoniker, config)
+                : null;
+        }
+
+        internal static readonly Command StartNoServiceCmd = RegisterCommand(
+            name: "start-noservice",
+            description: @"Starts a server process without a backing symbol service client (useful for testing/pinging the daemon).",
+            needsIpcClient: false,
+            clientAction: (conf, _) =>
+            {
+                var symbolConfig = new SymbolConfig(string.Empty, new Uri("file://xyz"));
+                var daemonConfig = CreateDaemonConfig(conf);
+                var vsoClientTask = TaskSourceSlim.Create<ISymbolClient>();
+                vsoClientTask.SetException(new NotSupportedException());

```

Comment:
This looks like a very strange pattern.. Should have have a helper for taht to says `NotSupportedSymbolClient`?

Reply:
Followed the existing practice in DropDaemon (the exception is used here to ensure that the client does not call into AS endpoint)

---
In reply to: [313488953](https://github.com/microsoft/BuildXL/pull/727#discussion_r313488953) [](ancestors = 313488953)